### PR TITLE
Make nbf check configurable

### PR DESCRIPTION
**Pull request by techaman-m**
## Proposed changes

Added claims set by creating an enum `RequiredClaims` and changed method signature `verifyJWT` to accept required claims as a list of strings.

### Why did it change

- [KBV-347](https://govukverify.atlassian.net/browse/KBV-347)

### CODE REVIEW

File: lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AccessTokenService.java [hunk 1]
```diff
@@ -143,7 +145,10 @@ public TokenRequest validateTokenRequest(
                             tokenRequest.getClientAuthentication().getClientID().getValue());
             SignedJWT signedJWT = privateKeyJWT.getClientAssertion();
 
-            jwtVerifier.verifyJWT(clientAuthenticationConfig, signedJWT);
+            jwtVerifier.verifyJWT(
+                    clientAuthenticationConfig,
+                    signedJWT,
+                    Arrays.asList(RequiredClaims.EXP.value, RequiredClaims.SUB.value));
```
Comment (jkunle) [inline lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AccessTokenService.java:151]: Why use `Array.asList` why not `List.of` same for all the occurences of you've used `Array.asList`

Comment (jkunle) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AccessTokenService.java:151]: Please can you remove ` var configurationServiceLocal = new ConfigurationService();` on line 63 and move `this.configurationService = new ConfigurationService();` to position 63

Comment (jkunle) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AccessTokenService.java:151]: I think you should remove    49 and 58 and also `new ListUtil()` shouldn't be in the constructor it is only used once and should be declared in the method where it is used

Comment (techaman-m) [reply lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AccessTokenService.java]: Addressed all comments

**Review by KaraokeStu — APPROVED** [review]
Looks good

**Pull request merged**